### PR TITLE
README - Added Toolchain section

### PR DESCRIPTION
**Pull request by amraboelela**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,3 +24,29 @@ Our first tasks for this project are:
 0. Incrementally add functionality back in.
 
 Some C headers and sources (e.g. `Availability.h`, `Block.h`, and the libclosure `runtime.c`) are similar to ones embedded into the CoreFoundation part of [swift-corelibs-foundation](http://github.com/apple/swift-corelibs-foundation). We should figure out a mechanism to share these instead of duplicating them across projects.
+
+## Toolchain
+To add libdispatch to the toolchain in Linux, you need to add libdispatch and install-libdispatch lines to ./swift/utils/build-presets.ini under `[preset: buildbot_linux]` section. Also to make the build run faster, you can comment the test lines. After those updates the section would be as following:
```
Comment (MadCoder) [inline README.md]: to the Swift toolchain in Linux (libdispatch can be built outside of swift too, so please be more precise).

**Comment by MadCoder** [comment]
+ @dgrove-oss  or @seabaylea  or anyone involved with the Linux Swift port to proofread this please

**Comment by amraboelela** [comment]
I actually got this info from @seabaylea page in github:
https://gist.github.com/seabaylea/cad7808615c52cfd2fc9d1debcad832f

**Comment by MadCoder** [comment]
I'm not disputing that, but since I never build for linux or almost, I'd rather have someone who does review, that's the point of the review process :)

**Comment by dgrove-oss** [comment]
Got back from vacation today; will dryrun instructions by tomorrow.  At first glance they look like they will work.  

However, do we want this level of detail in README.md?  The INSTALL file has similar (and more detailed) information about building a Swift toolchain that includes libdispatch using `utils/build-script --libdispatch -- --install-dispatch`.  Perhaps README.md should have an explicit pointer to INSTALL for building/install instructions (since this PR suggests that it isn't obvious where to find the information).

**Comment by amraboelela** [comment]
It is ok to have this info in INSTALL and point to it from README.md however can we change INSTALL file to be INSTALL.md (mark down format) just like README.md ?

**Comment by MadCoder** [comment]
I agree this should live in INSTALL

**Comment by dgrove-oss** [comment]
@MadCoder: any objection to having @amraboelela  include changing INSTALL to INSTALL.md when the pull request is updated?

**Comment by MadCoder** [comment]
nope, fine with me

**Comment by amraboelela** [comment]
Does that mean I have to do the change and then make another pull request?

**Comment by seabaylea** [comment]
Yeah, I agree that this fits better in INSTALL, which should become INSTALL.md. In fact, the whole document could do with refactoring/cleanup. 

In terms of the content, I would remove the information on commenting out the tests, and just state that (currently) you have to add `libdispatch` and `install-libdispatch` to the `build-presets.ini`. That should go away shortly once there's libdispatch is fully integrated into the toolchain.

@amraboelela You can either do the work in your existing branch and squash your commits, or create a new PR.

**Comment by amraboelela** [comment]
Done. Please check.

**Comment by MadCoder** [comment]
@seabaylea please review, and if it feels ok @amraboelela  please squash in a single commit

**Comment by amraboelela** [comment]
I was trying to do the squash but I failed :)
Any help is appreciated

**Comment by dgrove-oss** [comment]
Here's my "goto cheatsheet" for squashing commits: https://github.com/ginatrapani/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit

**Comment by amraboelela** [comment]
Created new pr #136

**Pull request closed without merging**